inspirehep/dojson/hep/fields/bd01x09x.py

```diff
@@ -55,36 +57,26 @@ def isbns2marc(self, key, value):
 def dois(self, key, value):
     """Other Standard Identifier."""
     value = utils.force_list(value)
-    out = []
+    value = strip_empty_values(value)
+    dois = self.get('dois', [])
     for val in value:
         if val:
-            if isinstance(val.get("2"), list):
-                if val.get("2", '')[0].lower() == "doi":
-                    if isinstance(val.get('a'), list):
-                        for v in val.get('a'):
-                            out.append({
-                                'value': v,
-                                'source': val.get('9')
-                            })
-                    else:
-                        out.append({
-                            'value': val.get('a'),
-                            'source': val.get('9')
+            for k, v in val.iteritems():
+                val[k] = utils.force_list(v)
+            if val.get("2") and val.get("2", '')[0].lower() == "doi":
+                for v in val.get('a'):
+                    if val.get('9'):
```

Comment:
This is actually added because we have records where the value is like this:
````
[{'a': ['10.1103/PhysRevLett.19.139'], '2': ['DOI']}]
```
so when it tries to do
```
'source': val.get('9', [''])[0]
```
It will give
```
IndexError: string index out of range
```
I will check if can solve it in a different way